libs/SalesforceSDKCore/SalesforceSDKCore/Classes/NativeLoginManagerInternal.swift

```diff
@@ -210,5 +222,342 @@ public class NativeLoginManagerInternal: NSObject, NativeLoginManager {
         let hash = SHA256.hash(data: data)
         return urlSafeBase64Encode(data: hash.dataRepresentation)
     }
-}
+    
+    // MARK: Headless, Password-Less Login Via One-Time-Passcode
+    
+    public func submitOtpRequest(
+        username: String,
+        reCaptchaToken: String,
+        otpVerificationMethod: OtpVerificationMethod) async throws -> OtpRequestResult
+    {
+        
+        // Validate parameters.
+        if !isValidUsername(
+            username: username.trimmingCharacters(in: .whitespacesAndNewlines)
+        ) {
+            return OtpRequestResult(nativeLoginResult: .invalidUsername)
+        }
+        
+        /*
+         * Create the OTP request body with the provided parameters. Note: The
+         * `emailtemplate` parameter isn't supported here, but could be added in
+         * the future.
+         */
+        // Determine the reCAPTCHA parameter for non-enterprise reCAPTCHA
+        let reCaptchaParameter: String? = if (isReCaptchaEnterprise) {
+            nil
+        } else {
+            reCaptchaToken
+        }
+        // Determine the reCAPTCHA "event" parameter for enterprise reCAPTCHA
+        let reCaptchaEventParameter: OtpRequestBodyReCaptchaEvent? = if (isReCaptchaEnterprise) {
+            OtpRequestBodyReCaptchaEvent(
+                token: reCaptchaToken,
+                siteKey: try reCaptchaSiteKeyId ?? { throw NSError(
```

Comment:
From a code-style perspective, this default-to-throwing-closure style is something I've used before since it's something of a call-site/in-expression guard and reads a bit like the Elvis `?:` operator in Kotlin.  I'd be happy to use whatever style rule we have in place, though.

I thought about adding specific cases for each of these errors as cases in `NativeLoginResult`.  I really wanted to provide detailed developer feedback about why their parameters were wrong, however.  So I went for a throw with a detailed and developer-friendly message.  If `NativeLoginResult` were a Swift enum and could have the message as an associated value, that would be the ideal.  We have Objective-C compatibility blocking that - alas!

Thoughts? After running the code a bit, I thought it might even be nice to retrofit the `.unknownError`, `.invalidPassword` and `.invalidUsername` cases to a similar thrown error.  The existing code outside of native login seems to mix and match error control styles, so I wasn't sure we had a hard style in place.